core/src/io/anuke/mindustry/input/InputHandler.java

```diff
@@ -806,16 +807,50 @@ public void drawArrow(Block block, int x, int y, int rotation, boolean valid){
     void iterateLine(int startX, int startY, int endX, int endY, Cons<PlaceLine> cons){
         Array<Point2> points;
         boolean diagonal = Core.input.keyDown(Binding.diagonal_placement);
+
         if(Core.settings.getBool("swapdiagonal") && mobile){
             diagonal = !diagonal;
         }
 
+        if(block instanceof PowerNode){
+            diagonal = !diagonal;
+        }
+
         if(diagonal){
             points = Placement.pathfindLine(block != null && block.conveyorPlacement, startX, startY, endX, endY);
         }else{
             points = Placement.normalizeLine(startX, startY, endX, endY);
         }
 
+        if(block instanceof PowerNode){
+            Array<Point2> skip = new Array<>();
+            final int[] i = {0};
+            points.each(req -> {
```

Comment:
Why not use a standard for-loop here? It's less lines, and avoids ugly array wrappers.

Reply:
guess it was just a remnant that stuck from something that worked 😅